src/compiler/time.ts

```diff
@@ -40,21 +40,21 @@ export function formula(timeUnit, fieldRef) {
   return fn + '(' + fieldRef + ')';
 }
 
-export function maxLength(timeUnit, encoding: Encoding) {
-  switch (timeUnit) {
+export function maxLength(fieldDef, encoding: Encoding) {
+  switch (fieldDef.timeUnit) {
     case 'seconds':
     case 'minutes':
     case 'hours':
     case 'date':
       return 2;
     case 'month':
+      if (fieldDef.axis.abbreviatedTimeNames)
+        return 3;
```

Comment:
I guess so. We don't really have a way to do this properly because it may depend on the locale setting.